pyodide-build/pyodide_build/out_of_tree/venv.py

```diff
@@ -0,0 +1,258 @@
+                sys.exit(main())
+            """
+        )
+    )
+    (venv_bin / "pip").chmod(0o777)
+
+    pyversion = f"{sys.version_info.major}.{sys.version_info.minor}"
+    other_pips = [
+        venv_bin / "pip3",
+        venv_bin / f"pip{pyversion}",
+        venv_bin / f"pip-{pyversion}",
+    ]
+
+    for pip in other_pips:
+        pip.unlink()
+        pip.symlink_to(venv_bin / "pip")
+
+
+def create_pyodide_script(venv_bin: Path) -> None:
+    """Write pyodide cli script into the virtualenv bin folder"""
+    import os
+
+    # The pyodide cli must be invoked outside of the virtual env. In order to
+    # ensure this, we have to restore VIRTUAL_ENV, PATH, and PYODIDE_ROOT to
+    # their current values before invoking it..
+    # In case we are inside an xbuildenv, make sure to make the path relative to
+    # this file and not relative to PYODIDE_ROOT.
+    pyodide_build_path = str(Path(__file__).parents[2])
+
+    # Resetting PATH and VIRTUAL_ENV will temporarily restore us to the virtual
+    # environment that 'pyodide venv' was invoked in (or no virtual environment
+    # if we aren't currently in one). To be extra careful, we set PYTHON_PATH too.
+    environment_vars = [f"PYTHONPATH={pyodide_build_path}"]
+    for environment_var in ["VIRTUAL_ENV", "PATH", "PYODIDE_ROOT"]:
+        value = os.environ.get(environment_var, "''")
+        environment_vars.append(f"{environment_var}={value}")
+    environment = " ".join(environment_vars)
+
+    pyodide_path = venv_bin / "pyodide"
+    pyodide_path.write_text(
+        dedent(
+            f"""
+            #!/bin/sh
+            {environment} exec {sys.executable} -m pyodide_build.out_of_tree "$@"
+            """
+        )
+    )
+    pyodide_path.chmod(0o777)
+
+
+def install_stdlib(venv_bin: Path) -> None:
+    """Install micropip and all unvendored stdlib modules"""
+    # Micropip we can install with pip!
+    toload = ["micropip"]
+    result = subprocess.run(
+        [venv_bin / "pip", "install", *toload],
+        capture_output=True,
+        encoding="utf8",
+    )
+    check_result(result, "ERROR: failed to invoke pip")
+
+    # Other stuff we need to load with loadPackage
+    # TODO: Also load all shared libs.
+    result = subprocess.run(
+        [
+            venv_bin / "python",
+            "-c",
+            dedent(
+                """
+                from _pyodide._importhook import UNVENDORED_STDLIBS_AND_TEST;
+                from pyodide_js import loadPackage;
+                loadPackage(UNVENDORED_STDLIBS_AND_TEST);
+                """
+            ),
+        ],
+        capture_output=True,
+        encoding="utf8",
+    )
+    check_result(result, "ERROR: failed to install unvendored stdlib modules")
+
+
+def create_pyodide_venv(dest: Path) -> None:
+    """Create a Pyodide virtualenv and store it into dest"""
+    print("Creating Pyodide virtualenv at", str(dest))
+    from virtualenv import session_via_cli  # type: ignore[import]
```

Comment:
Yeah this use case is out of scope for `venv` because it only creates virtual environments for the interpreter it is run in. So `python-exe -m venv` makes a virtual environment for `python-exe`. Since `venv` won't work correctly in Pyodide, it isn't really useful. virtualenv handles it without any trouble because it takes an interpreter parameter.

If it makes you feel any better, nox also uses virtualenv:
https://github.com/wntrblm/nox/blob/ffb4fca2b81a5905bf8dae674b33a8371da1c5f5/pyproject.toml#L30